### ISSUE TITLE
Run custom commands before to build

### ISSUE THREAD

**Issue by uilianries** [conan-io/conan-package-tools#136]
Hi!

When I run my jobs using Docker, sometimes I need to install some specific package by APT (e.g. subversion), or execute some non-trivial script. The problem is that all docker commands executed to build/test/upload my packages are provided by package tools. So, it's not possible to add any additional entrypoint before build.

For now, I need to create custom docker images to add new layers including my dependencies.

My suggestion is support some custom script to be executed as *entrypoint*:

```python
import os
from conan.packager import ConanMultiPackager

if __name__ == "__main__":
    builder = ConanMultiPackager(entrypoint=os.path.join("entrypoint", "install.py") # could be .sh
    builder.add_common_builds()
    builde.run()
```

When `entrypoint` is present, the script on path is executed once by `self.run()`, before to create temporary profile or create the packages.

Regards!

**Comment by solvingj** [comment]
In `build.py`, perhaps try `from conans import SystemPackageTool` and use `install` as you would in a `conanfile`. I haven't tested, just a thought.

**Comment by uilianries** [comment]
it won't work. All commands are executed on host, in our case, Travis host image.

When we execute build.run(), conan-package-tools invokes docker and apply all jobs. However, these events are totally protected from external commands.

Seems like this:

```
$ python build.py
  ...
  docker run -n conangcc6 lasote/conangcc6
  docker exec conangcc6 /bin/bash -c "conan profile create temp ..."
  docker exec conangcc6 /bin/bash -c "conan create package_ref -p temp ..."
```

**Comment by solvingj** [comment]
Oh, of course, i see

**Comment by memsharded** [comment]
Maybe it is important to clarify the naming, use something like ``docker_entry_script``, because it is not just some script that it is launched always (is it used in Windows?). For other use cases, a simple ``os.system()`` can be used, of course.

**Comment by uilianries** [comment]
Good point. This case is only to be executed on Docker images. `docker_entry_script` could be a good name.

**Comment by lasote** [comment]
Released 0.9.0 with #143
